### DISCUSSION TITLE
[announcer] Bluesky

### DISCUSSION THREAD

**Discussion by TomCools** [jreleaser/jreleaser#1379] [Ideas]
Hi,

I've been working on a Bluesky ("new Twitter") Java client recently, so got some experience with how that API looks.
If you are interested, I'm willing to add a Bluesky announcer. I would **not** add a dependency on my client library but I would instead create an SDK in this repository (like you've done for other announcers) and only implement the methods needed to post to Bluesky.

Caveat, I do not yet have access to Bluesky , but I would like to enlist some other members of the BeJUG to help out on this (some of them DO have accounts on Bluesky already). 

Kind regards,
Tom

**Comment by aalmiray** [comment]
That'd be great @TomCools! If possible, the sdk implementation should use the same/similar dependencies as the Mastodon announcer, that is, Feign and Apache HttpClient if needed.

**Comment by TomCools** [comment]
![image](https://user-images.githubusercontent.com/11362563/235960929-b69aebf7-ff72-4ee7-b74c-2394d3634edd.png)

Already recruited @SimonVerhoeven.

**Reply by aalmiray** [reply]
Some notes for implementing this feature:
- read-only models go in `api/jreleaser-model-api`.
- read/write model goes in `core/jreleaser-model-impl`.
- validation is also on `core/jreleaser-model-impl`.
- don't forget to hook the announcer in `JReleaserSupport`.
- gradle model goes in `plugins/jreleaser-gradle-plugin`.
- sdk goes in `sdks/jreleaser-bluesky-java-sdk`.

I'd recommend following the trail of the `MastodonAnnouncer`.

**Comment by TomCools** [comment]
We added the Draft PR: https://github.com/jreleaser/jreleaser/pull/1380 for an initial review.
